MovingBlocks/Terasology
# Compile against Java 17

### Contains
This pull request updates the game to build and run under Java 17. It increases the minimum required JDK version to 17.

Both JUnit and Mockito in `engine-tests` have been updated since the older versions would not run under Java 17.

### How to test
- The game should start-up without crashing.
- You should also be able to create a new world and start playing in it.
- You should be able to place and destroy blocks within the world.
- The game should play as before.

### Notes
- This depends on MovingBlocks/Logistics#9 and MovingBlocks/JenkinsAgentPrecachedJava#4 in a complicated fashion. Both of those pull requests technically depend on this pull request, creating a cyclic dependency. Manually pushing the initial builder image may be needed to resolve this.

Closes #3976
